src/NHibernate.Test/NHSpecificTest/GH0831/FixtureByCode.cs

```diff
@@ -0,0 +1,201 @@
+		}
+
+		[Test]
+		public void CanHandleEquals()
+		{
+			CanFilter(e => decimal.Equals(e.EntityValue, 1.0m));
+			CanFilter(e => decimal.Equals(1.0m, e.EntityValue));
+		}
+
+		[Test]
+		public void CanHandleFloor()
+		{
+			AssumeFunctionSupported("floor");
+
+			CanFilter(e => decimal.Floor(e.EntityValue) > 1.0m);
+			CanSelect(e => decimal.Floor(e.EntityValue));
+		}
+
+		[Test]
+		public void CanHandleMultiply()
+		{
+			CanFilter(e => decimal.Multiply(e.EntityValue, 10m) > 10m);
+			CanFilter(e => decimal.Multiply(10m, e.EntityValue) > 10m);
+
+			CanSelect(e => decimal.Multiply(e.EntityValue, 10m));
+			CanSelect(e => decimal.Multiply(10m, e.EntityValue));
+		}
+
+		[Test]
+		public void CanHandleNegate()
+		{
+			CanFilter(e => decimal.Negate(e.EntityValue) > -1.0m);
+			CanSelect(e => decimal.Negate(e.EntityValue));
+		}
+
+		[Test]
+		public void CanHandleRemainder()
+		{
+			CanFilter(e => decimal.Remainder(e.EntityValue, 2) == 0);
+			CanFilter(e => decimal.Remainder(2, e.EntityValue) < 1);
+
+			CanSelect(e => decimal.Remainder(e.EntityValue, 2));
+			CanSelect(e => decimal.Remainder(2, e.EntityValue));
+		}
+
+		[Test]
+		public void CanHandleRound()
+		{
+			AssumeFunctionSupported("round");
+
+			CanFilter(e => decimal.Round(e.EntityValue) >= 2.0m);
+			CanFilter(e => decimal.Round(e.EntityValue, 1) >= 1.5m);
+
+			// SQL round() always rounds up. 
+			CanSelect(e => decimal.Round(e.EntityValue), entities.Select(e => decimal.Round(e.EntityValue, MidpointRounding.AwayFromZero)));
+			CanSelect(e => decimal.Round(e.EntityValue, 1), entities.Select(e => decimal.Round(e.EntityValue, 1, MidpointRounding.AwayFromZero)));
+		}
+
+		[Test]
+		public void CanHandleSubtract()
+		{
+			CanFilter(e => decimal.Subtract(e.EntityValue, 1m) > 1m);
+			CanFilter(e => decimal.Subtract(2m, e.EntityValue) > 1m);
+
+			CanSelect(e => decimal.Subtract(e.EntityValue, 1m));
+			CanSelect(e => decimal.Subtract(2m, e.EntityValue));
+		}
+
+		private void CanFilter(Expression<Func<Entity, bool>> predicate)
+		{
+			using (ISession session = OpenSession())
+			using (session.BeginTransaction())
+			{
+				IEnumerable<Entity> inMemory = entities.Where(predicate.Compile()).ToList();
+				IEnumerable<Entity> inSession = session.Query<Entity>().Where(predicate).ToList();
+
+				CollectionAssert.AreEquivalent(inMemory, inSession);
+			}
+		}
+
+		private void CanSelect(Expression<Func<Entity, decimal>> predicate)
+		{
+			IEnumerable<decimal> inMemory = entities.Select(predicate.Compile()).ToList();
+
+			CanSelect(predicate, inMemory);
+		}
+
+		private void CanSelect(Expression<Func<Entity, decimal>> predicate, IEnumerable<decimal> expected)
+		{
+			using (ISession session = OpenSession())
+			using (session.BeginTransaction())
+			{
+				IEnumerable<decimal> inSession = session.Query<Entity>().Select(predicate).ToList();
+
+				Assert.That(inSession.OrderBy(x => x), Is.EqualTo(expected.OrderBy(x => x)).Within(0.001M));
```

Comment:
I would rather avoid the order by on result values, which may hide a trouble (granted, unlikely). Instead, the `inSession` query could be completed by an `.OrderBy(e => e.EntityValue)` before the `Select`, which should ensure the db results have been computed on a list having the same ordering than the in-memory one.